qiskit/transpiler/runningpassmanager.py

```diff
@@ -120,25 +92,33 @@ def _normalize_flow_controller(self, flow_controller):
                 raise TranspilerError('The flow controller parameter %s is not callable' % name)
         return flow_controller
 
-    def run(self, circuit):
+    def run(self, circuit, output_name=None, callback=None):
         """Run all the passes on a QuantumCircuit
 
         Args:
             circuit (QuantumCircuit): circuit to transform via all the registered passes
-
+            output_name (str): The output circuit name. If not given, the same as the
+                               input circuit
+            callback (callable): A callback function that will be called after each pass execution.
         Returns:
             QuantumCircuit: Transformed circuit.
         """
         name = circuit.name
         dag = circuit_to_dag(circuit)
         del circuit
 
+        if callback:
+            self.callback = callback
```

Comment:
i don't get how callback works in this RunningPassManager class. you just set `self.callback`, and nothing else seems to happen.

Reply:
This makes possible to set a callback at passmanager running-time.